plugin-name/phpstan.neon

```diff
@@ -1,64 +1,53 @@
-        - '#Argument of an invalid type object supplied for foreach, only iterables are supported\.#'
-        - '#Access to an undefined property object::\$is_search\.#'
-        - '#Access to an undefined property object::\$pending\.#'
-        - '#Access to an undefined property object::\$ID\.#'
-        - '#Call to an undefined method object::set\(\)\.#'
-        - '#Call to an undefined method object::get\(\)\.#'
-        # sanitize_text_field is used always with wp_unslash that based on the input, the output is the same type
-        - '/^Parameter #1 \$str of function sanitize_text_field expects string, array<string>\|string given\.$/'
-        # https://core.trac.wordpress.org/ticket/43304
-        - '/^Parameter #2 \$deprecated of function load_plugin_textdomain expects string, false given\.$/'
-        # WP-CLI accepts a class as callable
-        - '/^Parameter #2 \$callable of static method WP_CLI::add_command\(\) expects callable\(\): mixed, \S+ given\.$/'
-        # Stubs for WP-CLI are not available
-        - '#on an unknown class WP_CLI\.#'
-        # void as typehint is available only on PHP 7.1+
-        - '#has no return typehint specified\.#'
-        # CMB2 is not detected by PHPStan
-        - '#cmb2#'
-        - '#Cmb2#'
-        # Freemius is not detect by PHPStan
-        - '#fs_dynamic_init#'
-        # Uninstall is not detect by PHPStan
-        - '#pn_uninstall#'
-        # Other method to silence that PHPStan cannot detect
-        - '#Call to an undefined method object::is_#'
-        - "#Cannot access offset 'id' on object.#"
-        - '#Access to an undefined property object::\$query.#'
-        - '#invalid type array<WP_Term>\|int supplied for foreach#'
-        - '#Function is_amp_endpoint not found.#'
+        # TODO Add file names!!
+
+        # Void as typehints are available only on PHP 7.1+
+        - '#has no return typehint specified\.$#'
+        # In wp_count_posts(): $counts = (object) $counts;
+        - '#^Access to an undefined property object::\$pending\.$#'
+        # AMP plugin stubs can be also generated
+        - '#^Function is_amp_endpoint not found\.$#'
+        # PHPStan is better safe than sorry!
+        # https://github.com/phpstan/phpstan/issues/1060#issuecomment-667675767
+        - '#^Argument of an invalid type object supplied for foreach, only iterables are supported\.$#'
+        # https://github.com/WPBP/WordPress-Plugin-Boilerplate-Powered/issues/162
+        - '#^Parameter \#2 \$object_id of function cmb2_metabox_form expects int, string given\.$#'
+        # https://github.com/Freemius/wordpress-sdk/pull/439
+        - '#^Parameter \#1 \$module of function fs_dynamic_init expects array<string, string>, array<string, array<string, string>\|bool\|string> given\.$#'
+        # TODO Any can POST an array in a $_POST item, May I send a PR?
+        - '#^Parameter \#1 \$str of function sanitize_text_field expects string, array<string>\|string given\.$#'
```

Comment:
Any can POST an array in a $_POST item, May I send a PR?

Reply:
Well that is a WP function, where do you want to do a pr, to the boilerplate?

Reply:
> to the boilerplate?

Yes. Anyone can manipulation his/her browser and send an **array* in a e.g. `$_POST['var']` - so not a string but an array, it is an HTML feature.

Reply:
wp_unsplash detects string and arrays and outputs the same type as the input:

- string for strings
- array for arrays <-- This can be abused!

Reply:
Ok let's do a pr, I think that will be need just for ImpExp as I can remember.